File: app/src/main/java/com/chartapp/bottom/Bottom.java
package com.chartapp.bottom;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.MenuItem;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;
import androidx.appcompat.widget.Toolbar;
import androidx.core.content.ContextCompat;
import androidx.fragment.app.Fragment;

import com.chartapp.R;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.google.firebase.messaging.FirebaseMessaging;


public class Bottom extends AppCompatActivity  {


    Toolbar toolbar;
    BottomNavigationView bottom_nav;
    private static final String TAG = "MyActivity";






    @SuppressLint("ResourceAsColor")
    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_bottom2);
        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
        bottom_nav = findViewById(R.id.bottomNavView);
        bottom_nav.setOnNavigationItemSelectedListener(navListener);
        getSupportFragmentManager().beginTransaction()
                .replace(R.id.fragment_layout,new ChartFragment()).commit();
        toolbar = findViewById(R.id.toolbar_main);
        setSupportActionBar(toolbar);
        toolbar.setBackgroundDrawable(new ColorDrawable(getResources().getColor(R.color.toolbar)));
        getWindow().setStatusBarColor(ContextCompat.getColor(this,R.color.toolbar));
        fcm();



    }





    private BottomNavigationView.OnNavigationItemSelectedListener navListener = new BottomNavigationView.OnNavigationItemSelectedListener() {
        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
            Fragment selectedFragment = null;

            switch (item.getItemId()){
                case R.id.chart_menu:
                    selectedFragment = new ChartFragment();
                    break;
                case R.id.soft_menu:
                    selectedFragment = new SoftFragment();
                    break;
                case R.id.other_menu:
                    selectedFragment = new OtherFragment();
                    break;
            }
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.fragment_layout,selectedFragment).commit();
            return true;
        }
    };

public void fcm(){
    FirebaseMessaging.getInstance().getToken()
            .addOnCompleteListener(new OnCompleteListener<String>() {
                @Override
                public void onComplete(@NonNull Task<String> task) {
                    if (!task.isSuccessful()) {
                         Toast.makeText(getApplicationContext(), "Fetching FCM registration token failed",Toast.LENGTH_SHORT).show();
                        return;
                    }
                    String token = task.getResult();
                }
            });
}

    @Override
    protected void onDestroy() {
        toolbar = null;
        bottom_nav = null;
        super.onDestroy();
    }
}

File: app/src/main/java/com/chartapp/bottom/Bottom.kt
package com.chartapp.bottom

import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.bottomnavigation.BottomNavigationView
import android.annotation.SuppressLint
import android.os.Bundle
import com.chartapp.R
import androidx.appcompat.app.AppCompatDelegate
import com.chartapp.bottom.ChartFragment
import android.graphics.drawable.ColorDrawable
import androidx.core.content.ContextCompat
import com.chartapp.bottom.SoftFragment
import com.chartapp.bottom.OtherFragment
import com.google.firebase.messaging.FirebaseMessaging
import com.google.android.gms.tasks.OnCompleteListener
import android.widget.Toast
import androidx.appcompat.widget.Toolbar
import androidx.fragment.app.Fragment

class Bottom : AppCompatActivity() {
    lateinit var toolbar: Toolbar
    lateinit var bottom_nav: BottomNavigationView
    @SuppressLint("ResourceAsColor")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_bottom2)
        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
        bottom_nav = findViewById(R.id.bottomNavView)
        bottom_nav.setOnNavigationItemSelectedListener(navListener)
        supportFragmentManager.beginTransaction()
            .replace(R.id.fragment_layout, ChartFragment()).commit()
        toolbar = findViewById(R.id.toolbar_main)
        setSupportActionBar(toolbar)
        toolbar.setBackgroundDrawable(ColorDrawable(resources.getColor(R.color.toolbar)))
        window.statusBarColor = ContextCompat.getColor(this, R.color.toolbar)
        fcm()
    }

    private val navListener = BottomNavigationView.OnNavigationItemSelectedListener { item ->
        var selectedFragment: Fragment? = null
        when (item.itemId) {
            R.id.chart_menu -> selectedFragment = ChartFragment()
            R.id.soft_menu -> selectedFragment = SoftFragment()
            R.id.other_menu -> selectedFragment = OtherFragment()
        }
        supportFragmentManager.beginTransaction()
            .replace(R.id.fragment_layout, selectedFragment!!).commit()
        true
    }

    fun fcm() {
        FirebaseMessaging.getInstance().token
            .addOnCompleteListener(OnCompleteListener { task ->
                if (!task.isSuccessful) {
                    Toast.makeText(
                        applicationContext,
                        "Fetching FCM registration token failed",
                        Toast.LENGTH_SHORT
                    ).show()
                    return@OnCompleteListener
                }
                val token = task.result
            })
    }



    companion object {
        private const val TAG = "MyActivity"
    }
}